packages/react-ai/src/components/AIConversation/views/Controls/InputControl.tsx

```diff
@@ -0,0 +1,198 @@
+    </svg>
+  ),
+  stroke: '#0D1A26',
+  ...baseIconProps,
+});
+
+const AttachFileIcon = withBaseElementProps(Icon, attachIconProps);
+
+const AttachFileContainer = withBaseElementProps(View, {
+  className: `${INPUT_BLOCK}__${ATTACH_FILE_BLOCK}__container`,
+});
+
+const VisuallyHidden = withBaseElementProps(View, {
+  className: `${INPUT_BLOCK}__visually-hidden`,
+});
+
+const AttachFileButton = withBaseElementProps(Button, {
+  'aria-label': 'Attach file',
+  className: `${INPUT_BLOCK}__button ${INPUT_BLOCK}__button--attach`,
+});
+
+const sendIconProps = () => ({
+  children: (
+    <svg>
+      <g clipPath="url(#clip0_387_1426)">
+        <path
+          d="M14.3199 7.98214L1.66669 1.23486L3.97674 7.98214M14.3199 7.98214L1.66669 14.5689L3.97674 7.98214M14.3199 7.98214H3.97674"
+          strokeWidth="1.5"
+          strokeLinecap="square"
+          strokeLinejoin="round"
+        />
+      </g>
+      <defs>
+        <clipPath id="clip0_387_1426">
+          <rect width="100%" height="100%" />
+        </clipPath>
+      </defs>
+    </svg>
+  ),
+  stroke: 'white',
+  ...baseIconProps,
+});
+
+const SendIcon = withBaseElementProps(Icon, sendIconProps);
+
+const SendButtonBase = withBaseElementProps(Button, {
+  'aria-label': 'Send message',
+  className: `${INPUT_BLOCK}__button ${INPUT_BLOCK}__button--send`,
+  disabled: false,
+});
+
+const SendButton: typeof SendButtonBase = React.forwardRef(function SendButton(
+  { disabled, onClick, ...rest },
+  ref
+) {
+  // TODO should come from context
+  const isWaitingForResponse = false;
+  // TODO send message on click
+  return (
+    <SendButtonBase
+      disabled={isWaitingForResponse}
+      onClick={() => {}}
+      ref={ref}
+      {...rest}
+    />
+  );
+});
+
+const TextInputBase = withBaseElementProps(TextArea, {
+  className: `${INPUT_BLOCK}__input`,
+  placeholder: 'Message Raven',
+  id: `${INPUT_BLOCK}-text-input`,
+});
+
+const TextInput: typeof TextInputBase = React.forwardRef(function TextInput(
+  { onChange, placeholder, ...rest },
+  ref
+) {
+  const [input, setInput] = React.useState('');
+
+  // TODO should come from context or prop
+  const isFirstMessage = false;
+
+  React.useEffect(() => {
+    const textarea = document.getElementById(`${INPUT_BLOCK}-text-input`);
+
+    const handleResize = () => {
+      if (textarea) {
+        textarea.style.height = 'auto';
+        textarea.style.height = `${textarea.scrollHeight}px`;
+      }
+    };
+
+    if (textarea && textarea instanceof HTMLTextAreaElement) {
+      textarea.addEventListener('input', handleResize);
+      handleResize();
+    }
+
+    return () => {
+      if (textarea) {
+        textarea.removeEventListener('input', handleResize);
+      }
+    };
+  }, [input]);
+
+  return (
+    <TextInputBase
+      data-testid="text-input"
+      onChange={(e) => setInput(e.target.value)}
+      {...(isFirstMessage ? { placeholder: 'Ask anything...' } : {})}
+      {...rest}
```

Comment:
This placement of `rest` props still overwrites the values declared above it which could potentially break functionality. A simple way to prevent these issues is to always lead with the `rest` before all declared `props`

Reply:
Ok gotcha, moved up